components/header-bar/src/header-bar.stories.js

```diff
@@ -278,3 +278,23 @@ WithLastOnlineInfo.parameters = {
         },
     },
 }
+
+export const WithOnLogoutCallback = args => (
+    <Provider config={mockConfig}>
+        <CustomDataProvider data={customData}>
+            <HeaderBar {...args} />
+        </CustomDataProvider>
+    </Provider>
+)
+WithOnLogoutCallback.args = {
+    onLogout: () => {
+        console.log('onLogout called')
+        // tests callback async handling
+        return new Promise(res =>
+            setTimeout(() => {
+                res()
+                console.log('onLogout resolved')
+            }, 2000)
+        )
+    },
```

Comment:
This verifies that navigation waits for onLogout to resolve. Loading mask and cypress test coming soon